### PR TITLE
feat(ontology): 2.5.0 polyglot descriptors + IOntologySource + AONT037 (PR-A)

### PR DESCRIPTION
**Pull request by rsalus**
## Summary

First half of the Strategos 2.5.0 polyglot ingestion slice. Ships the descriptor schema, the source extension point, and the compile-time invariant — independently mergeable to unblock basileus on `MartenOntologySource`.

Closes (partially): #37 (`IOntologySource`+provenance), #48 (polyglot descriptors), #43 (AONT037 trigger; AONT200-series in PR-B).

**Design:** `docs/designs/2026-05-10-ontology-2-5-0-polyglot-ingestion.md`
**Plan:** `docs/plans/2026-05-10-ontology-2-5-0-polyglot-ingestion.md` (Tasks 1–22)

## What ships

| DR | Surface | Tasks |
|---|---|---|
| DR-1 | Polyglot descriptor schema (`ClrType?`, `SymbolKey`, `SymbolFqn`, `LanguageId` on Object/Link/Property) | 1–4 |
| DR-2 | AONT037 `PolyglotInvariantViolated` source-generator diagnostic | 18, 19 |
| DR-3 | `IOntologySource` interface + `OntologyOptions.AddSource<T>` DI | 7, 12 |
| DR-4 | `OntologyDelta` sealed-abstract record + 8 variants | 8 |
| DR-5 | `IOntologyBuilder.ObjectTypeFromDescriptor` + `ApplyDelta` (all 8 variants + unknown-variant guard) | 9, 10, 11 |
| DR-6 | `DescriptorSource` provenance + `MergeTwo` lattice (identity + property + link union) | 1, 14, 15 |
| DR-9 | `TestOntologySource` fixture + Roslyn `GetDocumentationCommentId` round-trip | 20, 22 |
| DR-10 | `OntologyCompositionException` aggregates diagnostics; source-error propagation with `SourceId`; AONT205 at delta-apply | 5, 16, 17 |
| (integration) | `OntologyGraphBuilder.Build()` drains registered sources; synthetic merge/diagnostic matrix | 13, 21 |

**Out of scope (PR-B):** AONT200-series graph-freeze diagnostics (AONT201–AONT208), AONT041 retarget, `OntologyValidateTool.FindObjectType` retarget. PR-B begins once this merges.

## Test plan

- [x] `dotnet build src/strategos.sln` — 0 warnings, 0 errors
- [x] `Strategos.Ontology.Tests`: 750/750 (+84 from baseline 666)
- [x] `Strategos.Ontology.Generators.Tests`: 86/86 (+8 from baseline 78)
- [x] `Strategos.Ontology.MCP.Tests`: 123/123 (unchanged)

## Notable deviations from the plan / design

1. **Identity-invariant guard is asymmetric** — enforced in `ObjectTypeDescriptor.SymbolKey` init setter only, not in the parameterless ctor. `new ObjectTypeDescriptor { Name=..., DomainName=... }` without either identity field compiles; AONT037 catches that case at the source-generator level, which is the design's intended detection layer.
2. **`AddSource<T>` is an instance method on `OntologyOptions`**, not an extension method — matches the surrounding `AddDomain<T>` pattern. Captures `static () => new T()` factories with `[DynamicallyAccessedMembers(PublicConstructors)]` for trim safety; avoids pulling `Microsoft.Extensions.DependencyInjection` (non-Abstractions) into `Strategos.Ontology`.
3. **`OntologyBuilder.ObjectTypeFromDescriptor` performs in-place cross-provenance `MergeTwo` fold** when an existing descriptor occupies the `(DomainName, Name)` slot with opposite provenance. Order-independent — works whether hand or ingested arrives first. Required for the Task 21 end-to-end matrix.
4. **AONT037 analyzer is purely syntactic** — matches `ObjectType` non-generic invocations with no `symbolKey:` named arg and no `typeof(...)` argument. Avoids semantic dependency on the descriptor-by-name overload (which doesn't ship in this PR).
5. **Roslyn `SymbolKey` substituted with `ISymbol.GetDocumentationCommentId()`** in Task 22 — Roslyn's `SymbolKey` is `internal`; cref ID is the stable public alternative and is what SCIP ingesters canonicalize on.
6. **`OntologyDelta` shell shipped in Task 7 commit**, variants filled in Task 8 — `IOntologySource` references `OntologyDelta` in its method signatures, so the shell had to exist for the Task 7 contract test to compile. Variant behavior is still pinned by Task 8's `OntologyDeltaTests`.
7. **`VSTHRD002` suppression on `OntologyGraphBuilder.DrainSources`** — bridges `IAsyncEnumerable` to sync `Build()` via `Task.Run + GetAwaiter().GetResult()`. Documented inline; live-invalidation via `SubscribeAsync` is the v2.6.0 path that returns the surface to fully async.

## Stack

PR-A of 2 in the Strategos 2.5.0 polyglot ingestion slice. PR-B follows once this merges.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/designs/2026-05-10-ontology-2-5-0-polyglot-ingestion.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+  IOntologySource.LoadAsync     →  OntologyDelta[]
+                                       ↓
+                                 IOntologyBuilder.ApplyDelta
+                                       ↓
+                                 Two-input fold (MergeTwo, ADR §9.1)
+                                       ↓
+                                 Graph-freeze with AONT200-series checks
+                                       ↓
+                                 OntologyGraph (consumed by PR #59 surfaces)
+```
```
[the remark this answers is not in the document]
Comment (rsalus) [reply docs/designs/2026-05-10-ontology-2-5-0-polyglot-ingestion.md]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: added the  language tag on the ASCII architecture fence.

File: docs/designs/2026-05-10-ontology-2-5-0-polyglot-ingestion.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+}
+```
+
+`OntologyGraphBuilder` drains `LoadAsync` from each registered source at construction. `SubscribeAsync` is a v2.5.0 surface — Strategos ships the consumer but does not wire live invalidation in this slice; consumers may complete the async enumerable immediately. Live invalidation lands when basileus exercises the surface.
+
+**Acceptance criteria:**
+- `IOntologySource` defined in `Strategos.Ontology` namespace, public.
+- `OntologyBuilderOptions.AddSource<T>()` extension method registers `T : IOntologySource` as transient.
+- `OntologyGraphBuilder.Build()` drains all registered sources' `LoadAsync` before returning the graph.
+- A `TestOntologySource` test fixture exists in `Strategos.Ontology.Tests.TestInfrastructure`.
```
[the remark this answers is not in the document]
Comment (rsalus) [reply docs/designs/2026-05-10-ontology-2-5-0-polyglot-ingestion.md]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: rewrote the source-drain paragraph to clarify  drains  as part of graph composition (after the hand-authored  pass, before validation). The at construction wording is gone — design and acceptance criteria now agree.

File: docs/plans/2026-05-10-ontology-2-5-0-polyglot-ingestion.md [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+## PR-A acceptance gate
+
+Run before opening PR-A:
+
+```
+dotnet build strategos.sln  # 0 warnings, 0 errors
+dotnet test src/Strategos.Ontology.Tests/Strategos.Ontology.Tests.csproj
+dotnet test src/Strategos.Ontology.Generators.Tests/Strategos.Ontology.Generators.Tests.csproj
+dotnet test src/Strategos.Ontology.MCP.Tests/Strategos.Ontology.MCP.Tests.csproj
+```
```
[the remark this answers is not in the document]
Comment (rsalus) [reply docs/plans/2026-05-10-ontology-2-5-0-polyglot-ingestion.md]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: language-tagged the pre-PR command fence as .

File: src/Strategos.Ontology.Tests/Integration/RoslynSymbolKeyIntegrationTests.cs [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                public string Id { get; set; } = "";
+                public decimal Amount { get; set; }
+            }
+        }
+        """;
+
+    [Test]
+    [SkipIfRoslynIntegrationDisabled]
+    public async Task RoslynSymbolKey_RoundTripThroughBuilder_PreservesIdentity()
+    {
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology.Tests/Integration/RoslynSymbolKeyIntegrationTests.cs:71]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: added a one-line XML `<summary>` on `RoslynSymbolKey_RoundTripThroughBuilder_PreservesIdentity` describing the round-trip contract the test guards.

File: src/Strategos.Ontology/Builder/OntologyBuilder.cs [hunk 5]
```diff
@@ -43,4 +47,267 @@ public ICrossDomainLinkBuilder CrossDomainLink(string name)
...
+        {
+            case OntologyDelta.AddObjectType add:
+                ValidateIngestedIntentInvariant(add.Descriptor);
+                ObjectTypeFromDescriptor(add.Descriptor);
+                break;
+
+            case OntologyDelta.UpdateObjectType update:
+                ValidateIngestedIntentInvariant(update.Descriptor);
+                ApplyUpdateObjectType(update);
+                break;
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/Builder/OntologyBuilder.cs:184]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: `UpdateObjectType` now routes through the same DR-6 cross-provenance fold (`TryCrossProvenanceMerge` → `MergeTwo.Merge`) when the existing and incoming descriptors carry opposite provenance. Same-provenance Updates still replace at the existing index. The Update tests split into `ApplyDelta_UpdateObjectType_SameProvenance_OverwritesExisting` and `ApplyDelta_UpdateObjectType_CrossProvenance_FoldsThroughMergeTwo`.

File: src/Strategos.Ontology/Configuration/OntologyOptions.cs [hunk 6]
```diff
@@ -17,6 +17,7 @@ public sealed class OntologyOptions
     private readonly List<WorkflowMetadataBuilder> _workflowMetadata = [];
     private readonly List<Action<IServiceCollection>> _serviceRegistrations = [];
     private readonly List<(int Order, Func<IServiceProvider, IActionDispatcher, IActionDispatcher> Factory)> _dispatcherDecorators = [];
+    private readonly List<Func<IOntologySource>> _sourceFactories = [];
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/Configuration/OntologyOptions.cs:20]: Acknowledged; declining for v2.5.0 with a documented limitation. `AddOntology` composes the graph **before** `BuildServiceProvider()` runs — by the time we drain `LoadAsync` the real DI container does not yet exist. The `new()` constraint is the trim/AOT-safe activator for that bootstrap path; the parallel `services.AddTransient<IOntologySource, T>` registration is the forward-compat surface for v2.6.0 live invalidation (`SubscribeAsync`), which resolves fresh source instances from the real container after it is built. DI-injected sources at compose time will need a `Func<IServiceProvider, T>` overload backed by a bootstrap scope; tracked for v2.6.0 alongside `SubscribeAsync`. Test sources today (`TestOntologySource`) and the basileus `MartenOntologySource` design plug in via the `IOntologySource[]`/`AddSources(...)` surface directly, so the parameterless restriction is not load-bearing for v2.5.0 consumers.

File: src/Strategos.Ontology/Descriptors/ObjectTypeDescriptor.cs [hunk 7]
```diff
@@ -1,10 +1,120 @@
...
+        {
+            _symbolKey = value;
+            if (_clrType is null && _symbolKey is null)
+            {
+                throw new InvalidOperationException(
+                    "ObjectTypeDescriptor violates the DR-1 identity invariant: "
+                    + "at least one of ClrType or SymbolKey must be non-null. "
+                    + "Hand-authored descriptors must supply ClrType; ingested descriptors must supply SymbolKey.");
+            }
+        }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/Descriptors/ObjectTypeDescriptor.cs:82]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: added an `EnsureIdentityInvariant` boundary check at `OntologyBuilder.ObjectTypeFromDescriptor` and `ApplyUpdateObjectType`. The `SymbolKey` init setter still catches the explicit-set bypass; the boundary check catches the parameterless-initializer bypass (`new ObjectTypeDescriptor { Name = ..., DomainName = ... }`) so every descriptor reaching the composed graph satisfies DR-1.

File: src/Strategos.Ontology/Internal/OntologyGraphHasher.cs [hunk 8]
```diff
@@ -359,13 +359,13 @@ private static void WriteWorkflowChains(BinaryWriter writer, OntologyGraph graph
...
-                              .ThenBy(w => w.ProducedType.ClrType.FullName ?? string.Empty, StringComparer.Ordinal))
+                              .ThenBy(w => w.ConsumedType.ClrType?.FullName ?? w.ConsumedType.SymbolKey ?? string.Empty, StringComparer.Ordinal)
+                              .ThenBy(w => w.ProducedType.ClrType?.FullName ?? w.ProducedType.SymbolKey ?? string.Empty, StringComparer.Ordinal))
         {
             writer.Write("W|");
             WriteString(writer, w.WorkflowName);
-            WriteString(writer, w.ConsumedType.ClrType.FullName ?? string.Empty);
-            WriteString(writer, w.ProducedType.ClrType.FullName ?? string.Empty);
+            WriteString(writer, w.ConsumedType.ClrType?.FullName ?? w.ConsumedType.SymbolKey ?? string.Empty);
+            WriteString(writer, w.ProducedType.ClrType?.FullName ?? w.ProducedType.SymbolKey ?? string.Empty);
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/Internal/OntologyGraphHasher.cs]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: replaced `?? string.Empty` with `?? descriptor.Name` on both `WorkflowChain` tie-breaker columns. `Name` is `required` on every descriptor, so the fallback can no longer collapse distinct types to the same hash input.

File: src/Strategos.Ontology/Merge/MergeTwo.cs [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+            if (seen.Contains(p.Name))
+            {
+                continue;
+            }
+
+            result.Add(p with { Source = DescriptorSource.Ingested });
+        }
+
+        return result;
+    }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/Merge/MergeTwo.cs]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: `MergeProperties` and `MergeLinks` return their backing list via `.AsReadOnly()`, so consumers cannot downcast to `List<T>` and mutate merged state.

File: src/Strategos.Ontology/OntologyCompositionException.cs [hunk 10]
```diff
@@ -1,14 +1,98 @@
...
+    /// observed.
+    /// </summary>
+    public ImmutableArray<OntologyDiagnostic> NonFatalDiagnostics { get; }
+
     public OntologyCompositionException(string message)
         : base(message)
     {
+        Diagnostics = ImmutableArray<OntologyDiagnostic>.Empty;
+        NonFatalDiagnostics = ImmutableArray<OntologyDiagnostic>.Empty;
     }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/OntologyCompositionException.cs:45]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: added XML docs to the public ctors of `OntologyCompositionException` following StyleCop SA1642 ("Initializes a new instance of the `OntologyCompositionException` class...").

File: src/Strategos.Ontology/OntologyCompositionException.cs [hunk 11]
```diff
@@ -1,14 +1,98 @@
...
+    /// Constructs an exception aggregating one or more error-severity
+    /// diagnostics. The default message lists the first diagnostic's
+    /// identifier to make logs scannable.
+    /// </summary>
+    public OntologyCompositionException(ImmutableArray<OntologyDiagnostic> diagnostics)
+        : base(BuildMessage(diagnostics, ImmutableArray<OntologyDiagnostic>.Empty))
+    {
+        Diagnostics = diagnostics.IsDefault ? ImmutableArray<OntologyDiagnostic>.Empty : diagnostics;
+        NonFatalDiagnostics = ImmutableArray<OntologyDiagnostic>.Empty;
+    }
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/OntologyCompositionException.cs:71]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: rewrote the constructor summaries to the SA1642 phrasing ("Initializes a new instance of the ... class aggregating ...").

File: src/Strategos.Ontology/OntologyGraphBuilder.cs [hunk 12]
```diff
@@ -28,17 +29,54 @@ internal OntologyGraphBuilder AddWorkflowMetadata(IEnumerable<WorkflowMetadataBu
...
+    /// to be drained at <see cref="Build"/> time. Each source's
+    /// <c>LoadAsync</c> is iterated and emitted deltas applied to the
+    /// matching per-domain builder. Sources contribute before composition
+    /// so all subsequent validation observes a unified descriptor set.
+    /// </summary>
+    public OntologyGraphBuilder AddSources(IEnumerable<IOntologySource> sources)
+    {
+        ArgumentNullException.ThrowIfNull(sources);
+        _sources.AddRange(sources);
+        return this;
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/OntologyGraphBuilder.cs:87]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: `AddSources` now iterates the enumerable, rejecting null entries with an `ArgumentException` so the drain loop can always attribute exceptions/deltas to a stable `SourceId`.

File: src/Strategos.Ontology/OntologyGraphBuilder.cs [hunk 13]
```diff
@@ -868,6 +948,85 @@ private static bool TryResolveDomainKeyed(
...
+            }
+        }
+    }
+
+    private static async Task DrainSourceCoreAsync(
+        IOntologySource source,
+        Dictionary<string, OntologyBuilder> buildersByDomain)
+    {
+        await foreach (var delta in source.LoadAsync(CancellationToken.None))
+        {
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/OntologyGraphBuilder.cs:1566]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: added a non-breaking `Build(CancellationToken)` overload that threads cancellation through `DrainSources` and the `IAsyncEnumerable.WithCancellation` call on `LoadAsync`. The original `Build()` delegates to `Build(CancellationToken.None)` so callers that don't need cancellation are unchanged.

File: src/Strategos.Ontology/Sources/IOntologySource.cs [hunk 14]
```diff
@@ -0,0 +1,35 @@
+namespace Strategos.Ontology;
+
+/// <summary>
+/// Extension point enabling ontology graph contributions from sources
+/// beyond hand-authored <c>DomainOntology.Define()</c>. Registered via DI
+/// (see <c>OntologyBuilderOptions.AddSource&lt;T&gt;()</c>). Drained at
+/// startup by <see cref="OntologyGraphBuilder"/>.
```
[the remark this answers is not in the document]
Comment (rsalus) [reply src/Strategos.Ontology/Sources/IOntologySource.cs]: Fixed in b76d6e67efdf8926024138e9138b0d810f762671: corrected the XML doc reference to `OntologyOptions.AddSource<T>()` (matching the actual class) on `IOntologySource`. The same correction landed in the design doc's DR-3 acceptance criteria.

**Comment by rsalus** [comment]
## Shepherd iteration 1 — CodeRabbit review addressed

Fix commit: `b76d6e6` (PR-A) / `f52f921` (PR-B rebased on top).

All 14 inline threads have a per-thread reply. High-level disposition:

### Major findings (design-correctness)

| Finding | Disposition |
|---|---|
| `UpdateObjectType` overwrites opposite-provenance descriptor instead of merging (`Builder/OntologyBuilder.cs`) | **Fixed.** Update now routes through `TryCrossProvenanceMerge` → `MergeTwo` for cross-provenance pairs; same-provenance Updates still replace at the existing index. Tests split into same- and cross-provenance cases. |
| DR-1 identity invariant bypassable when both fields omitted (`Descriptors/ObjectTypeDescriptor.cs`) | **Fixed.** `EnsureIdentityInvariant` boundary check at `ObjectTypeFromDescriptor` + `ApplyDelta` closes the parameterless-initializer bypass; the `SymbolKey` init-setter check still catches the explicit-set case. |
| Empty-string identity fallback weakens deterministic hashing (`Internal/OntologyGraphHasher.cs`) | **Fixed.** Falls back to `descriptor.Name` (required, always non-null) so distinct types can no longer collapse to identical hash input. |
| `MergeProperties` / `MergeLinks` return mutable `List<T>` behind `IReadOnlyList<T>` (`Merge/MergeTwo.cs`) | **Fixed.** Results wrapped via `.AsReadOnly()`. |
| Uncancellable `LoadAsync(CancellationToken.None)` at startup (`OntologyGraphBuilder.cs`) | **Fixed.** Added non-breaking `Build(CancellationToken)` overload that threads cancellation through `DrainSources` and `IAsyncEnumerable.WithCancellation`. |
| `AddSource<T>` `new()` constraint blocks DI-injected sources (`Configuration/OntologyOptions.cs`) | **Declined for v2.5.0**, follow-up tracked. `AddOntology` composes the graph before `BuildServiceProvider()` runs — by drain time the real DI container does not exist. The `new()` constraint is the trim/AOT-safe activator for the bootstrap path; the parallel transient DI registration is the v2.6.0 forward-compat surface for `SubscribeAsync`. A `Func<IServiceProvider, T>` overload backed by a bootstrap scope is the v2.6.0 follow-up. |

### Minor findings (docs/style)

| Finding | Disposition |
|---|---|
| Architecture fence missing language tag (design doc §3) | **Fixed.** `text` tag added. |
| Source-drain timing contradiction (design doc DR-3 prose) | **Fixed.** Rewrote to say `Build()` drains `LoadAsync` as part of graph composition; "at construction" wording gone. |
| Command-fence missing `bash` (plan doc) | **Fixed.** |
| XML doc on `RoslynSymbolKey_RoundTripThroughBuilder_PreservesIdentity` (test) | **Fixed.** |
| `IOntologySource` XML doc references wrong class (`OntologyBuilderOptions`) | **Fixed.** Now `OntologyOptions`. Same correction applied in the design doc. |
| `OntologyCompositionException` ctors missing XML docs / SA1642 phrasing | **Fixed.** "Initializes a new instance of the ... class" phrasing on all four ctors. |
| `AddSources` no null-entry guard | **Fixed.** Iterates and rejects null entries with `ArgumentException`. |

### Validation

- `Strategos.Ontology.Tests` — 768 passed, 0 failed (this PR + PR-B rebased)
- `Strategos.Ontology.Generators.Tests` — 86 passed
- `Strategos.Ontology.MCP.Tests` — 123 passed
- `Strategos.Tests` — 790 passed
- `dotnet build src/strategos.sln` — 0 warnings, 0 errors

### Design intent + axiom:design check

- **DR-1 (identity invariant)**: now enforced both at the descriptor's init setters (explicit-set cases) and at the builder boundary (initializer-bypass case). No descriptor reaches the composed graph without `ClrType != null OR SymbolKey != null`.
- **DR-5 / DR-6 (Update fold)**: cross-provenance `UpdateObjectType` deltas now follow the same lateral lattice as the Add path, eliminating the asymmetry where hand intent could be silently overwritten by an ingested Update.
- **DR-10 (failure modes)**: `Build(CancellationToken)` makes the source-drain bounded; source-raised exceptions continue to surface with `SourceId` attribution.
- **DIM-1 (Topology)**: still single composition root; the cross-provenance fold remains in `MergeTwo` with one caller pattern (`TryCrossProvenanceMerge`).
- **DIM-2 (Observability)**: hash-input fallback now uses descriptor `Name` so unresolved chains carry a distinguishable identifier instead of an empty bucket.
- **DIM-7 (Resilience)**: cancellable drain removes the indefinite-startup hazard CodeRabbit flagged.

**Comment by rsalus** [comment]
@coderabbitai Full review.

**Comment by rsalus** [comment]
@coderabbitai Full review.

**Comment by rsalus** [comment]
## Iter 3: addressed review [#4302194834](https://github.com/lvlup-sw/strategos/pull/72#pullrequestreview-4302194834)

All 11 actionable items resolved in commit `135a010`. Evaluated each through axiom:design dimensions before applying — every item was a real defect (DIM-1 correctness or DIM-2 contract or DIM-4 doc accuracy), no over-engineering candidates.

### Correctness (DR-7 snapshot integrity)
- **`OntologyBuilder.Apply{Add,Rename,Remove}Property` + `Apply{Add,Remove}Link`** — mirror mutations to `_ingestedOriginals` via two small private helpers (`MutateIngestedOriginalProperties`, `MutateIngestedOriginalLinks`). No-op when the key has no ingested baseline.
- **`OntologyBuilder.ApplyUpdateObjectType`** — deferred `SyncIngestedOriginal` until after the merge decision. When an incoming hand-authored update folds against an existing ingested baseline, the snapshot is preserved (previously cleared, breaking AONT201–AONT208 diffs).
- **`MergeTwo.MergeProperties` / `MergeLinks`** — used `seen.Add(name)` (single-call test-and-add) on ingested append; duplicate ingested names now collapse to a single emission.
- **`TestOntologySource.LoadAsync` / `SubscribeAsync`** — `ThrowIfCancellationRequested` before the first `Task.Yield` so a pre-cancelled token fails fast even with empty deltas.

### Hardening
- **`OntologyBuilder.IngestedOriginals`** — returns a fresh `ReadOnlyDictionary` wrapper each call; consumers can no longer down-cast and mutate.
- **`OntologyDefinitionAnalyzer.ReportPolyglotInvariantViolations`** — requires the receiver type to be `IOntologyBuilder` (or `OntologyBuilder`) via `SemanticModel.GetTypeInfo`; accepts `clrType:` named arg as identity-satisfying regardless of expression form (typeof, variable, factory call).

### Doc / message accuracy
- **AONT037 `PolyglotInvariantViolated` message** — now references `builder.Object<T>()` / `clrType:` named arg, matching the actual registration API on `IOntologyBuilder`.
- **Design doc DR-3** — both `AddSource<T>` mentions corrected: it's an instance-method factory constructing via `new T()` (AOT-safe, no DI activation), not a DI transient registration. Note added that callers needing constructor injection should instantiate and register the instance themselves.
- **Design doc DR-1 AC** — replaced hardcoded `666 / 121` test counts with "all existing tests continue to pass".
- **`RoslynSymbolKeyIntegrationTests.SkipIfRoslynIntegrationDisabledAttribute`** — SA1642 constructor `<summary>` added.
- **`docs/plans/.../polyglot-ingestion.md` line 530** — `bash` language tag added to PR-B acceptance fence (MD040).

### Tests added (24 new cases)
- `IngestedOriginalsMirrorTests` (9): Add/Rename/Remove property and Add/Remove link snapshot mirroring; no-op when no baseline exists; cross-provenance fold snapshot preservation; hand-replaces-hand snapshot clearance; defensive wrapper invariant.
- `MergeTwoPropertyLinkTests`: duplicate ingested property + duplicate ingested link collapse regressions.
- `TestOntologySourceTests`: pre-cancelled empty-deltas Load case; pre-cancelled SubscribeAsync case.
- `AONT037AnalyzerTests`: `clrType:` named-arg no-fire; non-builder receiver no-fire.

### Verification
```
dotnet build src/strategos.sln  # 0 warnings, 0 errors
Strategos.Ontology.Tests         781 passed, 0 failed
Strategos.Ontology.Generators.Tests   89 passed, 1 skipped (deferred), 0 failed
Strategos.Ontology.MCP.Tests     126 passed, 0 failed
Strategos.Tests                  790 passed, 0 failed
```

No items skipped; no items deferred.

**Comment by rsalus** [comment]
@coderabbitai approve
